ksql-parser/src/main/java/io/confluent/ksql/parser/tree/BooleanLiteral.java

```diff
@@ -19,29 +19,30 @@
 import static java.util.Objects.requireNonNull;
 
 import com.google.common.base.Preconditions;
+import com.google.errorprone.annotations.Immutable;
 import java.util.Objects;
 import java.util.Optional;
 
-public class BooleanLiteral
-    extends Literal {
+@Immutable
+public class BooleanLiteral extends Literal {
 
   private final boolean value;
 
   public BooleanLiteral(final String value) {
     this(Optional.empty(), value);
   }
 
-  public BooleanLiteral(final NodeLocation location, final String value) {
-    this(Optional.of(location), value);
-  }
-
-  private BooleanLiteral(final Optional<NodeLocation> location, final String value) {
+  public BooleanLiteral(
+      final Optional<NodeLocation> location,
+      final String value
+  ) {
     super(location);
-    requireNonNull(value, "value is null");
-    Preconditions.checkArgument(
-        value.toLowerCase(ENGLISH).equals("true") || value.toLowerCase(ENGLISH).equals("false"));
+    this.value = requireNonNull(value, "value")
+        .toLowerCase(ENGLISH)
+        .equals("true");
 
-    this.value = value.toLowerCase(ENGLISH).equals("true");
+    Preconditions.checkArgument(value.toLowerCase(ENGLISH).equals("true")
```

Comment:
not for this PR but commenting anyway to track for myself: most SQL languages (e.g. [postgres](https://www.postgresql.org/docs/9.1/datatype-boolean.html)) allow for various aliases for boolean values

Reply:
Indeed